### DISCUSSION TITLE
Best Way To Replace Many Notes Programmatically

### DISCUSSION THREAD

**Discussion by XaverStiensmeier** [TriliumNext/Trilium#4634] [Q&A] [answered]
I am writing a little script that loads a bunch of web views. However, existing notes in the target folder should simply be overwritten in order to avoid the need to do so by hand. A simple way to do this would be to store their id as a label and then for every note before creating search for their label and delete any note found. Another way would be to just delete the entire parent note and recreate it or delete everything in the parent note.

I feel like this is a relatively common issue. Any "way to go" solutions for that issue?

# Backend Code

```js
// api url to access
var url = 'https://ilaris-online.de/api/ilaris/kreatur/';

async function fetchData() {
    // waits until promise becomes get response
    const response = await fetch(url);
    if (!response.ok) {
        api.log("Not OK!")
        throw new Error(`HTTP error! Status: ${response.status}`);
    }
    
    const kreaturen = await response.json();
    // for simplification we just use the first element
    api.log(kreaturen[0].name)
    // get folder holding all webViews later
    const kreaturenRoot = api.getNoteWithLabel('kreaturenRoot');
    // create new webView and set source accordingly using the label
    const resp = api.createNewNote({     
        parentNoteId: kreaturenRoot.noteId, 
        title: kreaturen[0].name,
        type: "webView",
        content: ''
    });
    resp.note.setLabel("webViewSrc", "https://ilaris-online.de/app/kreatur/" + kreaturen[0].id);
}

fetchData();
```

**Comment by WantToLearnJapanese** [comment]
Sorry, I don't understand why you need to update `webview` notes if they are pointing same URLs.

Anyway, your approach looks fine. But, You may not need an extra label as an ID since `webViewSrc` could be used as an ID.

**Reply by XaverStiensmeier** [reply]
Just to clarify: I need to execute it more than once because new database entries might be added in the future.

You are primarily voting for a: Check if it exists and if so, don't replace it?
It is not guaranteed that the URLs stay the same in the future as the database is still in an early state. So I feel like it's safer to simply replace everything.

**Reply by WantToLearnJapanese** [reply]
Ah, now I understand the situation.
Let me summarize the note structure.
For URLs `https://example.com/<path1>/<path2>/...`

* path1_1
  * path1_1(1)
  * path1_1(2)
* path1_2
  * ...

Is it right?
If it is right.
Let's say there is a `modification` on children of path1_1, then I will delete removed notes, and add newly added items as notes.

In short, I would check modifications from bottom to top(=descendants to ancestors).

Sorry if what I wrote is not clear. I am on the mobile device.

**Reply by XaverStiensmeier** [reply]
There's no real folder structure regarding the items. There is
- Kreaturen
    - Kreatur1
    - Kreatur2
 - Scripts
     - BackendCode
 - Other Nodes

I do not entirely understand what you mean by modification and delete removed notes. Modifications happen on the page, not locally. But it's fine that you need to trigger the update manually.

**Reply by WantToLearnJapanese** [reply]
I mean the modification of items retrieved from json.

**Reply by XaverStiensmeier** [reply]
I think checking if changes happened are overkill. Just default overwriting should be fast enough.

**Comment by zadam** [comment, marked as answer]
Just deleting an existing note and creating it again (like you have in your script) should be OK ...
